doc/source/ScriptingAPI/DictionarySyntax/index.md

```diff
@@ -0,0 +1,65 @@
+# Dictionary Syntax #
+
+Gaffer scripts use Python's dictionary syntax and notation to define the node graph. As you add to and manipulate the node graph, the dictionary builds a compendium of instances of node and plug classes. Each node is a dictionary key, with a node class as its value. Each plug is a sub-key of a node's key, with a plug class as its value. The following pseudocode shows a rough approximation of a node graph's dictionary:
+
+```python
+script = {
+    ["Node1"]:Module.NodeType1()
+    ["Node1"]["Plug1"]:Module.PlugType1()
+    ["Node1"]["Plug2"]:Module.PlugType2()
+
+    ["Node2"]:Module.NodeType2()
+    ["Node2"]["Plug1"]:Module.PlugType1()
+    ["Node2"]["in"]:GafferScene.ScenePlug( ["Node1"]["out"] )
+    
+    ["Node3"]:Module.NodeType3()
+    ["Node3"]["in"]:GafferScene.ScenePlug( ["Node2"]["out"] )
+}
+```
+
+When the script is evaluated, Gaffer traverses the dictionary, using the _in_ plugs of nodes to build a parent-child hierarchy. This becomes a dependency graph, as represented in the _Graph Editor_. At the start of all Gaffer node graphs is a root ScriptNode object, referenced by the `script` variable.
```

Comment:
This seems badly wrong - the in plugs are not used to build a parent-child hierarchy.  The connections between plugs define a dependency graph of the nodes, but nodes are not parented to each other by this process ( you can put one node actually inside another, but that has nothing to do with the in plugs ).  Or by parent-child hierarchy were you referring to the scene hierarchy?  That would seem highly confusing when we're in the middle of discussing how nodes and plugs are accessed.

Reply:
Oh, this makes it sound like the script variable is a global concept, but I think it's only related to the Script Editor window - a Gaffer session could contain several scripts, and only if you open a Script Editor attached to one would there be a script variable pointing to it.

Reply:
Will adjust. Re: the `script` variable, I was trying to indicate it was script-specific, but I should have used "each" instead of "all". Language is important. :stuck_out_tongue: